src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -5328,6 +5327,17 @@ private VisitArgumentResult VisitArgumentEvaluate(BoundExpression argument, RefK
                 case RefKind.In:
                     {
                         // Note: for lambda arguments, they will be converted in the context/state we saved for that argument
+                        if (conversion is { Kind: ConversionKind.ImplicitUserDefined })
```

Comment:
>if (conversion is { Kind: ConversionKind.ImplicitUserDefined }) [](start = 24, length = 63)

It's not clear to me why this logic is specific to argument conversions. Won't we have the same requirements for assignments?

Reply:
In an assignment, we know the target-type (if there is to be a user-defined conversion). That target-type doesn't change because of nullable analysis, so the user-defined conversion is still the right one.
In a call, the target-type comes from re-inferring the type arguments for the method as part of nullable analysis.